### DISCUSSION TITLE
Application.Current nullable?

### DISCUSSION THREAD

**Discussion by beto-rodriguez** [AvaloniaUI/Avalonia#9062] [Ideas]
I love that in the first preview of v11 you are using `nullable reference types` a lot.

But I have found at least one case where it seems unnecessary:

I think `Application.Current` must be of type `Application` instead of `Application?`, from my perspective it would be really strange to get a null reference from this property. Am I getting this wrong? is it posible for Avalonia users to get a null reference from this property?

![image](https://user-images.githubusercontent.com/10853349/193375121-7ffae1a7-82b9-4510-8a65-30346e9b3828.png)

**Comment by jp2masa** [comment]
`Application.Current` is null until you initialize and start the application.

**Reply by beto-rodriguez** [reply]
can we users of Avalonia use this property in an application that is/will not start?

**Reply by jp2masa** [reply]
Yes, for example try using it in `Main` before doing anything else, and it will be null:

```cs
        static void Main(string[] args)
        {
            Console.WriteLine(Application.Current is null);
            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
        }
```

**Reply by beto-rodriguez** [reply]
makes sense!
